namer/namer.cc

```diff
@@ -440,45 +440,69 @@ class NameInserter {
     }
 
     unique_ptr<ast::Expression> postTransformSend(core::MutableContext ctx, unique_ptr<ast::Send> original) {
-        core::SymbolRef method;
-        if (original->args.size() == 1) {
-            if (auto mdef = ast::cast_tree<ast::MethodDef>(original->args[0].get())) {
-                // this handles the `private def foo` case
-                method = mdef->symbol;
-            } else if (auto sym = ast::cast_tree<ast::Literal>(original->args[0].get())) {
-                // this handles the `private :foo` case
-                if (!sym->isSymbol(ctx)) {
-                    return original;
-                }
-                auto name = sym->asSymbol(ctx);
-                auto owner = ctx.owner.data(ctx)->enclosingClass(ctx);
-                if (original->fun._id == core::Names::privateClassMethod()._id) {
-                    owner = owner.data(ctx)->singletonClass(ctx);
-                }
-                method = ctx.state.lookupMethodSymbol(owner, name);
-                if (method == core::Symbols::noSymbol()) {
-                    return original;
-                }
-            } else {
```

Comment:
I moved most of this logic to a helper function so that i could call it recursively to implement my new case.

Reply:
(The diff looks better ignoring whitespace)